### ISSUE TITLE
Version fields on entity returned from persist() are undefined

### ISSUE THREAD

**Issue by arimus** [typeorm/typeorm#312]
This is causing problems when trying to immediately re-save the entity, since undefined + 1 == NaN.

Version: 0.0.8

```
Bootstrap(): failed to initialize database { Error: ER_BAD_FIELD_ERROR: Unknown column 'NaN' in 'field list'
    at Query.Sequence._packetToError (/Users/arimus/workspace/xxxxx/node_modules/mysql/lib/protocol/sequences/Sequence.js:52:14)
    at Query.ErrorPacket (/Users/arimus/workspace/xxxxx/node_modules/mysql/lib/protocol/sequences/Query.js:83:18)
    at Protocol._parsePacket (/Users/arimus/workspace/xxxxx/node_modules/mysql/lib/protocol/Protocol.js:280:23)
    at Parser.write (/Users/arimus/workspace/xxxxx/node_modules/mysql/lib/protocol/Parser.js:74:12)
    at Protocol.write (/Users/arimus/workspace/xxxxx/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (/Users/arimus/workspace/xxxxx/node_modules/mysql/lib/Connection.js:109:28)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:543:20)
    --------------------
    at Protocol._enqueue (/Users/arimus/workspace/xxxxx/node_modules/mysql/lib/protocol/Protocol.js:141:48)
    at PoolConnection.query (/Users/arimus/workspace/xxxxx/node_modules/mysql/lib/Connection.js:214:25)
    at /Users/arimus/workspace/xxxxx/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:213:49
    at new wrappedPromise (/Users/arimus/workspace/xxxxx/node_modules/async-listener/index.js:372:16)
    at MysqlQueryRunner.query (/Users/arimus/workspace/xxxxx/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:212:16)
    at MysqlQueryRunner.<anonymous> (/Users/arimus/workspace/xxxxx/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:265:51)
    at step (/Users/arimus/workspace/xxxxx/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:32:23)
    at Object.next (/Users/arimus/workspace/xxxxx/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:13:53)
    at /Users/arimus/workspace/xxxxx/node_modules/typeorm/driver/mysql/MysqlQueryRunner.js:7:71
    at new wrappedPromise (/Users/arimus/workspace/xxxxx/node_modules/async-listener/index.js:372:16)
  code: 'ER_BAD_FIELD_ERROR',
  errno: 1054,
  sqlState: '42S22',
  index: 0 }
```

**Comment by mattwilson1024** [comment]
Just come across the same issue myself. Is there a workaround?

**Comment by arimus** [comment]
I just re-checked to make sure, but this is still a problem in v0.0.11. I'm not sure if these commits to master made it into that release or not. If not, you can give them a try.

As for workaround options:

1. manually set a newly persisted objects version to 1 before trying to re-save it, since that's what you were going to get back anyways
2. re-fetch the object right after persist, at which point it will be properly populated (much heavier option, but I do this currently since I want to re-fetch all the relationships with properly populated IDs, etc.)

**Comment by pleerock** [comment]
This problem should be resolved in `0.2.0-alpha` versions now. I completely changed now version and create/update dates are generated - they are updated on the database side. You should not have this problem anymore. Try it in `npm i typeorm@next`
